packages/extension/src/contexts/LedgerContext/LedgerContext.tsx

```diff
@@ -751,6 +776,88 @@ const LedgerProvider: FC = ({ children }) => {
     [client, getCurrentWallet]
   );
 
+  const submitTransactionsBulk = useCallback(
+    async (payload: SubmitTransactionsBulkRequest) => {
+      const onError = payload.onError;
+      const wallet = getCurrentWallet();
+      if (!client) {
+        throw new Error('You need to be connected to a ledger');
+      }
+      if (!wallet) {
+        throw new Error('You need to have a wallet connected');
+      }
+
+      const processTransaction = async (txWithID: TransactionWithID) => {
+        const { ID, ...pl } = txWithID;
+
+        if (!pl.Account || txWithID.Account === '') {
+          pl.Account = wallet.publicAddress;
+        }
+
+        try {
+          // Validate the transaction
+          validate(pl as unknown as Record<string, unknown>);
+          // Prepare the transaction
+          const prepared: Transaction = await client.autofill(pl);
+          // Sign the transaction
+          const signed = wallet.wallet.sign(prepared);
+          // Submit the signed blob based on the mode
+          const tx = payload.noWait
+            ? await client.submit(signed.tx_blob)
+            : await client.submitAndWait(signed.tx_blob);
+
+          if (payload.noWait) {
+            // Throttle the requests
+            await new Promise((resolve) => setTimeout(resolve, 1000));
+          }
+
+          // Check transaction result
+          if (payload.noWait && !(tx as SubmitResponse).result.accepted) {
+            throw new Error("Couldn't submit the transaction");
+          }
+
+          if (
+            !payload.noWait &&
+            (!(tx as TxResponse).result.hash ||
+              ((tx as TxResponse).result.meta! as TransactionMetadata).TransactionResult !==
+                'tesSUCCESS')
+          ) {
+            throw new Error(
+              "Couldn't submit the signed transaction but the transaction was successful"
+            );
+          }
+
+          return {
+            ID,
+            hash: !payload.noWait ? (tx as TxResponse).result.hash : undefined,
+            accepted: payload.noWait ? true : undefined
+          };
+        } catch (e) {
+          Sentry.captureException(e);
+          return {
+            ID,
+            error: `Error while processing the transaction: ${JSON.stringify(e)}`
+          };
+        }
+      };
+
+      const results: TransactionBulkResponse[] = [];
+      for (const tx of payload.transactions) {
```

Comment:
This loop will send the requests one by one which is slow.
Have you thought about using `Promise.all` or `Promise.allSettled()` to send them in parallel and make the whole process faster?

Reply:
I tried with `Promise.all` but it was only really submitting the first transaction of each batch. My assumption was that I needed to throttle